wolfcrypt/src/rsa.c

```diff
@@ -3638,7 +3638,7 @@ static int RsaPrivateDecryptEx(const byte* in, word32 inLen, byte* out,
                     word32 j;
                     int start = (int)((size_t)pad - (size_t)key->data);
 
-                    for (j = 0; j < key->dataLen; j++) {
+                    for (j = 0; j < key->dataLen && i < outLen; j++) {
```

Comment:
Don't want to do this if possible - constant time code.
Line 3645 check 'i' against outLen.